airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -271,20 +281,39 @@ private void scanFile(MediaFile file, MusicFolder musicFolder, Date lastScanned,
         }
     }
 
-    private void updateGenres(MediaFile file, Genres genres) {
-        String genre = file.getGenre();
-        if (genre == null) {
-            return;
-        }
-        if (file.isAlbum()) {
-            genres.incrementAlbumCount(genre);
+    private List<Genre> getGenres(MediaFile file, Map<String, Genre> cache) {
```

Comment:
This method extracts the collection of genres from a media file parsing the genre ID3 tag. Ok but why this isn't done directly while the MediaFile object is instanciated ? This would simplify the library scan process. 

@devonbryant